File: backend/data-ingestion-server/src/main/java/com/insight_web/data_ingestion_server/adapters/kafka/KafkaQueueSender.java
package com.insight_web.data_ingestion_server.adapters.kafka;

import com.insight_web.data_ingestion_server.domain.TrackerEvent;
import com.insight_web.data_ingestion_server.usecase.TrackerEventProcessing;
import com.insight_web.data_ingestion_server.usecase.queueSender.QueueSender;
import lombok.RequiredArgsConstructor;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Service
public class KafkaQueueSender implements QueueSender {
    private static final String KAFKA_TOPIC = "insightweb-events";

    private final KafkaTemplate<String, TrackerEvent> kafkaTemplate;

    public KafkaQueueSender(KafkaTemplate<String, TrackerEvent> kafkaTemplate) {
        this.kafkaTemplate = kafkaTemplate;
    }

    @Override
    public void send(TrackerEvent event) {
        String key = event.getSiteId();
        kafkaTemplate.send(KAFKA_TOPIC, key, event);
    }
}

File: backend/data-ingestion-server/src/main/kotlin/com/insight_web/data_ingestion_server/adapters/kafka/KafkaQueueSender.kt
package com.insight_web.data_ingestion_server.adapters.kafka

import com.fasterxml.jackson.databind.ObjectMapper
import com.insight_web.data_ingestion_server.usecase.queueSender.QueueSender
import com.insight_web.domain.TrackingEvent
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service

@Service
class KafkaQueueSender(
    private val kafkaTemplate: KafkaTemplate<String, String>,
    private val objectMapper: ObjectMapper,
): QueueSender {
    private val KAFKA_TOPIC = "insightweb-events"

    override fun send(event: TrackingEvent) {
        kafkaTemplate.send(KAFKA_TOPIC, objectMapper.writeValueAsString(event))
    }
}